lib/xmerl/test/xmerl_SUITE.erl

```diff
@@ -696,3 +696,101 @@ datadir(Config) ->
 
 datadir_join(Config,Files) ->
     filename:join([datadir(Config)|Files]).
+
+%%======================================================================
+%% New formatter tests input/output
+%%======================================================================
+
+html() ->
+    "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\" \"DTD/xhtml1-strict.dtd\"><html>"
+    "<head><title>Doc Title</title><author>Ulf Wiger</author></head>"
+    "<h1>heading1</h1>"
+    "<p>This is a paragraph of text.</p>"
+    "<h2>heading2</h2>"
+    "<p>This is another paragraph.</p>"
+    "<table>"
+    "<thead><tr><td>head1</td><td>head2</td></tr></thead>"
+    "<tr><td>col11</td><td>col122</td></tr>"
+    "<tr><td>col21</td><td>col122</td></tr>"
+    "</table>"
+    "</html>".
+
+  html_indented() ->
+    "<?xml version=\"1.0\"?>"
+    "\n<html>"
+    "\n  <head>"
+    "\n    <title>Doc Title</title>"
+    "\n    <author>Ulf Wiger</author>"
+    "\n  </head>"
+    "\n  <h1>heading1</h1>"
+    "\n  <p>This is a paragraph of text.</p>"
+    "\n  <h2>heading2</h2>"
+    "\n  <p>This is another paragraph.</p>"
+    "\n  <table>"
+    "\n    <thead>"
+    "\n      <tr>"
+    "\n        <td>head1</td>"
+    "\n        <td>head2</td>"
+    "\n      </tr>"
+    "\n    </thead>"
+    "\n    <tr>"
+    "\n      <td>col11</td>"
+    "\n      <td>col122</td>"
+    "\n    </tr>"
+    "\n    <tr>"
+    "\n      <td>col21</td>"
+    "\n      <td>col122</td>"
+    "\n    </tr>"
+    "\n  </table>"
+    "\n</html>".
+
+  xml_namespace() ->
+    "<?xml version=\"1.0\"?>"
+    "<!-- initially, the default namespace is \"books\" -->"
+    "<book xmlns='urn:loc.gov:books' xmlns:isbn='urn:ISBN:0-395-36341-6'>"
+    "<title>Cheaper by the Dozen</title>"
+    "<isbn:number>1568491379</isbn:number>"
+    "<notes>"
+    "<!-- make HTML the default namespace for some comments -->"
+    "<p xmlns='urn:w3-org-ns:HTML'>"
+    "This is a <i>funny</i> book!"
+    "</p>"
+    "</notes>"
+    "</book>".
+
+  xml_namespace_indented() ->
+  "<?xml version=\"1.0\"?>"
+  "\n<book xmlns=\"urn:loc.gov:books\" xmlns:isbn=\"urn:ISBN:0-395-36341-6\">"
+  "\n  <title>Cheaper by the Dozen</title>"
+  "\n  <isbn:number>1568491379</isbn:number>"
+  "\n  <notes>"
+  "\n    <p xmlns=\"urn:w3-org-ns:HTML\">This is a <i>funny</i> book!</p>"
+  "\n  </notes>"
+  "\n</book>".
+
+  output_element_to_str(E) ->
+    Output = xmerl:export([E], xmerl_xml_indent),
+    [Str] = io_lib:format("~s", [lists:flatten(Output)]),
+    Str.
+
+  %%======================================================================
+  %% New formatter tests
+  %%======================================================================
+  'formatter-pass'(_Config) ->
+
+    FetchFun = fun(_DTDSpec, S) -> {ok, not_fetched, S} end,
+    % Generate based on namespace-example
```

Comment:
```suggestion
    %% Generate based on namespace-example
```
I believe line comments are supposed to start with `%%`, because of EMACS or sth.

Reply:
Fixed comments and renamed function to formatter_pass - to better fit the existing test-code.